uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/ACL/LeaveApproverTrait.php

```diff
@@ -1,24 +1,21 @@
 <?php
 
-/**
- * Access Control Lists Custom Query generator class
- *
- * This class is basically an Utility class used to generate queries needed for overriding addSelectWhereClause
- * for some specific BAO's
- *
- */
-class CRM_HRLeaveAndAbsences_API_Query_ACLQueryHelper {
+trait CRM_HRLeaveAndAbsences_ACL_LeaveApproverTrait {
+
+  use CRM_HRLeaveAndAbsences_Service_SettingsManagerTrait;
 
   /**
-   * This method returns a query string that limits contacts records that are available to a logged in user
-   * by taking into account user relationships. It checks whether the logged in user is a leave approver
-   * to some contacts or not. If yes the logged in user has access to contact records for the contacts/users he manages,
-   * If not the logged is limited to his/her own records only.
+   * This method returns a query string that limits contacts records that are
+   * available to a logged in user by taking into account user relationships.
+   * It checks whether the logged in user is a leave approver to some contacts
+   * or not. If yes the logged in user has access to contact records for the
+   * contacts/users he manages. If not the logged is limited to his/her own
+   * records only.
    *
    * @return string
    *   Query String
    */
-  public static function limitContactsByTakingUserRelationShipsIntoAccount() {
+  public function getLeaveApproverACLClauses() {
```

Comment:
This function name makes me think that the query only takes the leave approver into consideration whereas it also takes the leave contact into consideration too. i.e it checks based on whether its a leave approver or a leave contact. Perhaps we can give it a better name?